controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -359,6 +362,29 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 				if err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update StorageClass: %s", err)
 				}
+
+			case "VolumeSnapshotClass":
+				var volumeSnapshotClass *snapapi.VolumeSnapshotClass
+				data["csi.storage.k8s.io/snapshotter-secret-namespace"] = r.storageClassClaim.Namespace
+
+				if resource.Name == "cephfs" {
+					volumeSnapshotClass = r.getCephFSVolumeSnapshotClass(data)
+				} else if resource.Name == "ceph-rbd" {
+					volumeSnapshotClass = r.getCephRBDVolumeSnapshotClass(data)
+				}
+				storageClassClaimNamespacedName := r.getNamespacedName()
+
+				if annotations := volumeSnapshotClass.GetAnnotations(); annotations == nil {
+					volumeSnapshotClass.SetAnnotations(map[string]string{
+						StorageClassClaimAnnotation: storageClassClaimNamespacedName,
+					})
+				} else {
+					annotations[StorageClassClaimAnnotation] = storageClassClaimNamespacedName
+				}
```

Comment:
This is a repeating pattern that was observed in the past for labels as well. 
In that case, we introduced a util function to handle it (see here: https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storageclassclaim/storageclassclaim_controller.go#L955-L962) 

Can we do something similar and refactor this into
```suggestion
				r.addAnnotation(StorageClassClaimAnnotation, storageClassClaimNamespacedName)
```


This can also be used to replace the same code in the storageclassclass case